models/issue.go

```diff
@@ -1201,6 +1190,22 @@ func newIssue(e *xorm.Session, doer *User, opts NewIssueOptions) (err error) {
 
 // NewIssue creates new issue with labels for repository.
 func NewIssue(repo *Repository, issue *Issue, labelIDs []int64, assigneeIDs []int64, uuids []string) (err error) {
+	// Retry several times in case INSERT fails due to duplicate key for (repo_id, index); see #7887
+	i := 0
+	for {
+		if err = newIssueAttempt(repo, issue, labelIDs, assigneeIDs, uuids); !IsErrNewIssueInsert(err) {
+			// Usually err == nil
+			return err
+		}
```

Comment:
if err == nil {
  return
}

Reply:
@lunny have you noticed that the condition is `!IsErrNewIssueInsert(err)`? If you want me to open it it multiple lines, it will then be:
```
if err = newIssueAttempt(repo, issue, labelIDs, assigneeIDs, uuids); err == nil {
	return nil
}
if !IsErrNewIssueInsert(err) {
	return err
}
```
I just thought it was shorter. Should I change it anyway?

Reply:
@guillep2k I think the new one is correct but the old one is not correct even it was shorter.

Reply:
@lunny  OK, done.